content/docs/rfcs/64/README.md

```diff
@@ -0,0 +1,258 @@
+
+## Network shards
+
+Traffic in the Waku Network is sharded into eight [17/WAKU2-RLN-RELAY](https://rfc.vac.dev/spec/17/) pubsub topics.
+Each pubsub topic is named according to the static shard naming format
+defined in [51/WAKU2-RELAY-SHARDING](https://rfc.vac.dev/spec/51/)
+with `<shard_cluster_index>` set to `1` and
+`<shard_number>` occupying the range `0` to `7`.
+In other words, the Waku Network is a [17/WAKU2-RLN-RELAY](https://rfc.vac.dev/spec/17/) network
+routed on the combination of the eight pubsub topics:
+```
+/waku/2/rs/1/0
+/waku/2/rs/1/1
+...
+/waku/2/rs/1/7
+```
+
+## Roles
+
+There are two distinct roles evident in the network, those of:
+1) nodes, and
+2) applications.
+
+### Nodes
+
+Nodes are the individual software units
+using [10/WAKU2](https://rfc.vac.dev/spec/10/) protocols to form a p2p messaging network.
+Nodes, in turn, can participate in a shard as full relayers
+or by running a combination of protocols suitable for resource-restricted environments.
+These will be referred to as _relay nodes_ and
+_non-relay nodes_
+in the rest of this document.
+Nodes can also provide various services to the network,
+such as storing historical messages or protecting the network against spam.
+See the section on [default services](#default-services) for more.
+
+#### Relay nodes
+
+Relay nodes MUST follow [17/WAKU2-RLN-RELAY](https://rfc.vac.dev/spec/17/)
+to route messages to other nodes in the network
+for any of the pubsub topics [defined as the Waku Network shards](#network-shards).
+Relay nodes MAY choose to subscribe to any of these shards,
+but MUST be subscribed to at least one defined shard.
+Each relay node SHOULD be subscribed to as many shards as it has resources to support.
+If a relay node supports an encapsulating application,
+it SHOULD be subscribed to all the shards servicing that application.
+If resource restrictions prevent a relay node from servicing all shards used by the encapsulating application,
+it MAY choose to support some shards as a non-relay node.
+
+#### Bootstrapping and discovery
+
+Nodes MAY use any method to bootstrap connection to the network,
+but it is RECOMMENDED that each node retrieve a list of bootstrap peers to connect to using [EIP-1459 DNS-based discovery](https://eips.ethereum.org/EIPS/eip-1459).
+Relay nodes SHOULD use [33/WAKU2-DISCV5](https://rfc.vac.dev/spec/33/) to continually discover other peers in the network.
+Each relay node MUST encode its supported shards into its discoverable ENR
+as described in [51/WAKU2-RELAY-SHARDING](https://rfc.vac.dev/spec/51/#discovery).
+The ENR MUST be updated if the set of supported shards change.
+A node MAY choose to ignore discovered peers that do not support any of the shards in its own subscribed set.
+
+#### Default services
+
+For each supported shard,
+each relay node SHOULD enable and support the following protocols as a service node:
```

Comment:
Wondering if it would be better to mention run atleast 1 of these services along with relay Or is it better to start with all, so that we have more decentralization?

Reply:
I don't think we want to overdefine this in the spec. My intuition is that we want all nodes to provide all these services by default. The `SHOULD` indicates that there may indeed be reasons outside the spec for not configuring all services, but the impact must be understood.